packages/db/postgres/tables.sql

```diff
@@ -0,0 +1,154 @@
+  -- An error occurred unpinning.
+  'UnpinError',
+  -- The IPFS daemon has pinned the item.
+  'Pinned',
+  -- The IPFS daemon is currently pinning the item.
+  'Pinning',
+  -- The IPFS daemon is currently unpinning the item.
+  'Unpinning',
+  -- The IPFS daemon is not pinning the item.
+  'Unpinned',
+  -- The IPFS daemon is not pinning the item but it is being tracked.
+  'Remote',
+  -- The item has been queued for pinning on the IPFS daemon.
+  'PinQueued',
+  -- The item has been queued for unpinning on the IPFS daemon.
+  'UnpinQueued',
+  -- The IPFS daemon is not pinning the item through this CID but it is tracked
+  -- in a cluster dag
+  'Sharded'
+);
+
+-- An IPFS node that is pinning content.
+CREATE TABLE IF NOT EXISTS pin_location
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- Libp2p peer ID of the node pinning this pin.
+  peer_id         TEXT                                                          NOT NULL UNIQUE,
+  -- Name of the peer pinning this pin.
+  peer_name       TEXT,
+  -- Geographic region this node resides in.
+  region          TEXT
+);
+
+-- Information for piece of content pinned in IPFS.
+CREATE TABLE IF NOT EXISTS pin
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- Pinning status at this location.
+  status          pin_status_type                                               NOT NULL,
+  -- The content being pinned.
+  content_cid     TEXT                                                          NOT NULL REFERENCES content (cid),
+  -- Identifier for the service that is pinning this pin.
+  pin_location_id BIGINT                                                        NOT NULL REFERENCES pin_location (id),
+  inserted_at     TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  updated_at      TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  UNIQUE (content_cid, pin_location_id)
+);
+
+CREATE TYPE upload_type AS ENUM
+(
+  -- A CAR file upload.
+  'Car',
+  -- Files uploaded and converted into a CAR file.
+  'Upload',
+  -- A raw blob upload in the request body.
+  'Blob',
+  -- A multi file upload using a multipart request.
+  'Multipart'
+);
+
+-- An upload created by a user.
+CREATE TABLE IF NOT EXISTS upload
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- User that uploaded this content.
+  user_id         BIGINT                                                        NOT NULL REFERENCES user (id),
+  -- User authentication token that was used to upload this content.
+  -- Note: nullable, because the user may have used a Magic.link token.
+  auth_key_id     BIGINT REFERENCES auth_key (id),
+  -- The root of the uploaded content.
+  content_cid     TEXT                                                          NOT NULL REFERENCES content (cid),
+  -- CID for the content as provided by the user.
+  -- TODO: do we need this? Should we fail if they do not match?
+  source_cid      TEXT                                                          NOT NULL,
```

Comment:
is `content_cid` the base32 cidv1 normalised CID, while `source_cid` is the CID in the form we found in the CAR file, could be cidv0 etc?

Reply:
That is my assumption, this was grabbed from both nft.storage and https://github.com/web3-storage/web3-schema/blob/main/schema.sql#L54 . Not sure why to keep it though

Reply:
> is `content_cid` the base32 cidv1 normalised CID, while `source_cid` is the CID in the form we found in the CAR file, could be cidv0 etc?

yes, and content_cid is a FK into the content table

Reply:
@vasco-santos @alanshaw nft.storage is using `cid`  as the primary key for the content table... do we want to match that here? https://github.com/ipfs-shipyard/nft.storage/blob/7102ad5fc06d90a85027e40fa0aece3429b833e7/packages/api/db/tables.sql#L49

Reply:
Ok, I updated the comments to make them more clear. I think we should move with `cid` as the primary key yes! and we are already doing it